src/main/java/reposense/authorship/FileInfoExtractor.java

```diff
@@ -109,20 +110,36 @@
                 continue;
             }
 
+            if (!isValidAndNonBinaryFile(filePath, nonBinaryFilesList)) {
+                continue;
+            }
+
             if (Format.isInsideWhiteList(filePath, config.getFormats())) {
-                try {
-                    FileInfo currentFileInfo = generateFileInfo(config.getRepoRoot(), filePath);
-                    setLinesToTrack(currentFileInfo, fileDiffResult);
-                    fileInfos.add(currentFileInfo);
-                } catch (InvalidPathException ipe) {
-                    logger.warning(String.format(INVALID_FILE_PATH_MESSAGE_FORMAT, filePath));
-                }
+                FileInfo currentFileInfo = generateFileInfo(config.getRepoRoot(), filePath);
+                setLinesToTrack(currentFileInfo, fileDiffResult);
+                fileInfos.add(currentFileInfo);
             }
         }
 
         return fileInfos;
     }
 
+    /**
+     * Returns the list of non-binary files for the repo {@code repoConfig}.
+     */
+    public static Set<Path> getListOfNonBinaryFiles(RepoConfiguration repoConfig) {
```

Comment:
Is it necessary to use a `Set`, will there be duplicates?

Update the JavaDocs according. Returns a {@code Set<Path>} of non-binary files

Reply:
No there won't be duplicates. But set is used here for performance considerations, when repeatedly calling `isValidAndNonBinaryFile` to see if this set contains a specific file.

Will update JavaDocs accordingly